client/toDataByLocation.test.js

```diff
@@ -0,0 +1,87 @@
+const toDataByLocation = require('./toDataByLocation');
+
+test('Testing that an entry wont go through without approval or valid lat and lng', () => {
+    const fakeDataApproval = {
+        values: [
+            [],
+            ["approved", "city", "state", "lat", "lng"], // headers 
+            ["x", "Seattle", "WA", "47.6674625", "-122.3795306"], // valid entry
+            ["", "Burien", "CA", "57.4585642", "-100.3339504"], // invalid approval
+            ["x", "Seattle", "WA", "", "-122.3795306"], // invalid latitude 
+            ["x", "Seattle", "MO", "30", ""], // invalid longitude
+        ]
+    };    
+    Object.values(toDataByLocation(fakeDataApproval,'makers')).forEach((stateObj) => {
```

Comment:
the nested forEachs can be little difficult to read. I generally do something along the lines of:

```
const expected = {
  makers: {
    CA: {
      cities: [
        { name: 'Burien', entries: [ { lat: "57.4585642", lng: "-100.3339504", approved: 'x' } ] }
      ]
    }
  }
}

expect(toDataByLocation(fakeDataApproval, 'makers')).toEqual(expected);
```

(ignore that the example I gave was burien which shouldn't be in there 🤦) I like this style since it's really explicit about what you're expecting back.

Reply:
Oh I see.. that makes a lot of sense. I wasn't sure if I should mimic the structure of the expected output or pass in some fake data and check the output is right (what i was doing previously).

Reply:
that makes a lot of sense! Also, just checking, wouldn't the structure of the expected  output be
`  const expected = {
    "WA": {
      "cities": {
        "seattle": {
          "entries": [
            {"approved": "x", "city": "Seattle", "datasetKey": "makers", "lat": "47.6674625", "lng": "-122.3795306", "state": "WA"}
          ]
        }
      }
    }
  }`

Reply:
yes, totally. Sorry was just guessing off the top of my head!